pyang-apteryx-xml.py

```diff
@@ -353,6 +715,39 @@ def value_identityref(self, node, res):
                     if base_idref and hasattr(ch, "i_type_spec"):
                         self.value_attrib_identityref(res, ch)
 
+    def type_to_pattern(self, ntype):
+        """
+        Return a pattern for a given node type. Return None if
+        a pattern is not appropriate.
+        """
+        patt = None
+        if not isinstance(ntype, statements.Statement):
+            return None
+        if ntype.arg in ["int8", "int16", "int32", "uint8", "uint16", "uint32"]:
+            range = ntype.search_one("range")
+            if range is not None:
+                limits = range.arg.replace('..', '.').split('.')
+                l0 = int(limits[0])
+                l1 = int(limits[1])
+                if l0 > l1:
+                    l1, l0 = l0, l1
+                rfr = RegexForRange(l0, l1)
+                patt = f"{rfr}"
+            elif ntype.arg == "int8":
+                patt = "(-[1-9]|-12[0-7]|-?[1-9][0-9]|-?1[01][0-9]|[0-9]|12[0-8])"
```

Comment:
could these all be done with RegexForRange?

Reply:
Thanks for the comment. My new routine type_to_pattern was written to allow a recursive type to pattern type action due to the effects of unions. Since this deals with integer types and is effectively the same as parts of the sample_element routine it is called from, I decided to remove the integer pattern stuff from sample_element and call type_to_pattern instead. However, this leads me to think that if I removed the other types from sample_element and put them in type_to_pattern as well, I would make the code look even nicer. Would you like me to go that extra step? Are you OK with all ranges now being patterns in the final schemas?

Reply:
unfortunately we use the range keyword to identify that the value should be formated as a JSON integer ... hence we can't change existing integers to use patterns ... but it also means your data will probably be formatted incorrectly

Reply:
Ouch. The main reason for bringing this in is to allow a leaf-list which is a union of integer 1-4094 and strings representing a range "n..m" where n and m are in the range 1-4094. Would it be normal in this case to format the single integers as JSON integers or would everything revert to being a string? Is there potentially another way we could use to identify integers for JSON formatting purposes? I take it this means that every integer in every data model ends up with a range associated with it?